projects/igniteui-angular/src/lib/chips/themes/_base.scss

```diff
@@ -0,0 +1,412 @@
+                    border-color: color($color: 'warn', $variant: 800);
+                }
+            }
+    
+            &:hover {
+                @include e(item) {
+                    color: contrast-color($color: 'warn', $variant: 600);
+                    background: color($color: 'warn', $variant: 600);
+                    border-color: color($color: 'warn', $variant: 600);
+                }
+            }
+        }
+
+        @include m(danger) {
+            @include e(item) {
+                color: contrast-color($color: 'error', $variant: 500);
+                background: color($color: 'error', $variant: 500);
+                border-color: color($color: 'error', $variant: 500);
+            }
+
+            &:focus {
+                @include e(item) {
+                    color: contrast-color($color: 'error', $variant: 800);
+                    background: color($color: 'error', $variant: 800);
+                    border-color: color($color: 'error', $variant: 800);
+                }
+            }
+    
+            &:hover {
+                @include e(item) {
+                    color: contrast-color($color: 'error', $variant: 600);
+                    background: color($color: 'error', $variant: 600);
+                    border-color: color($color: 'error', $variant: 600);
+                }
+            }
+        }
+
+        @include e(content) {
+            @include ellipsis();
+
+            max-width: 32ch;
+
+            &:empty {
+                display: none;
+            }
+        }
+
+        @include e(select) {
+            display: inline-flex;
+            align-items: center;
+            max-width: rem(24px);
+            opacity: 1;
+            z-index: 1;
+            transition: max-width .12s $ease-in-out-quad, opacity .12s $ease-in-out-quad .06s;
+            transition-behavior: allow-discrete;
+
+            @starting-style {
+                max-width: 0;
+                opacity: 0;
+            }
+        }
+
+        @include e(remove) {
+            display: inline-flex;
+            color: var-get($_theme, 'remove-icon-color', currentColor);
+
+            &:empty {
+                display: none;
+            }
+
+            &:focus {
+                igx-icon {
+                    color: var-get($_theme, 'remove-icon-color-focus');
+                }
+            }
+        }
+
+        @include e(ghost) {
+            position: absolute;
+            box-shadow: var-get($_theme, 'ghost-elevation');
+            overflow: hidden;
+            color: var-get($_theme, 'focus-text-color');
+            background: var-get($_theme, 'ghost-background');
+            // If z-index is not set,
+            //the chip would hide behind it's parent grid as it has a z-index of 1
+            z-index: 10;
+
+            igx-avatar,
+            igx-circular-bar,
+            igx-icon {
+                --component-size: var(--chip-size);
+            }
```

Comment:
Why are these applied when the item is also a ghost, if we also set the sizes for the component explicitly in the code above?